### PR TITLE
feat: cursor shim for ~/.local/bin/cursor

### PR DESCRIPTION
**Pull request by ZanzyTHEbar**
Adds an optional `cursor` shim that delegates to `cursor-installer` when no Cursor binary is in PATH.

- **shim.sh**: Installed as `~/.local/bin/cursor`; finds real `cursor` in PATH, else runs `cursor agent` or `cursor-installer` with args.
- **scripts/ensure-shim.sh**: Idempotent install/update; skips if existing file is already our shim; never overwrites a non-shim binary.
- **install.sh**: Syncs shim + helper to `~/.local/share/cursor-installer`, runs ensure-shim after placing CLI.
- **cursor.sh**: Refreshes shim assets from GitHub on `--update`; runs ensure-shim before install/update.

Guards: only replace `~/.local/bin/cursor` when it is already a shim (shebang + marker) or missing.

### CODE REVIEW

File: shim.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+set -eu
+
+# Find cursor executable in PATH, excluding the current shim
+find_cursor() {
+  old_IFS="$IFS"
+  IFS=:
+  for dir in $PATH; do
+    [ -n "$dir" ] || continue
+    cursor_path="$dir/cursor"
+    if [ "$cursor_path" != "$HOME/.local/bin/cursor" ] && [ -x "$cursor_path" ]; then
```
[the remark this answers is not in the document]
Comment (ZanzyTHEbar) [reply shim.sh:11]: Hey! So after looking at this, I took some time to stress test my local setup and I WAS able to hit this edge-case. 

I will provide a fix for it. 

How I did it: 

- machine where cursor installer was used as primary mechanism
- install cursor cli
- run cursor installer AGAIN to overwrite cursor cli bash script entrypoint with shim 

This caused the shim to re-discover itself as the shim only used raw string comparison. 

I was then faced with a terminal "hang" as the shim re-executed itself.
